keras/src/saving/saving_lib.py

```diff
@@ -521,6 +522,10 @@ def save_weights_only(model, filepath, objects_to_skip=None):
             "Invalid `filepath` argument: expected a `.weights.h5` extension. "
             f"Received: filepath={filepath}"
         )
+
+    if file_utils.is_remote_path(filepath):
+        filepath = file_utils.get_file(model, filepath, cache_subdir="models")
```

Comment:
We should download it to a temporary filepath instead, and delete if after loading